delta/plugins/graph-analytics/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/graph/analytics/indexing/GraphAnalyticsStream.scala

```diff
@@ -0,0 +1,106 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.graph.analytics.indexing
+
+import cats.data.NonEmptyList
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.kernel.database.Transactors
+import ch.epfl.bluebrain.nexus.delta.plugins.graph.analytics.indexing.GraphAnalyticsResult.{Index, Noop, UpdateByQuery}
+import ch.epfl.bluebrain.nexus.delta.plugins.graph.analytics.model.JsonLdDocument
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.Files
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileState
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.resources.Resources
+import ch.epfl.bluebrain.nexus.delta.sdk.resources.model.ResourceState
+import ch.epfl.bluebrain.nexus.delta.sourcing.config.QueryConfig
+import ch.epfl.bluebrain.nexus.delta.sourcing.implicits._
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{ElemStream, EntityType, ProjectRef, Tag}
+import ch.epfl.bluebrain.nexus.delta.sourcing.offset.Offset
+import ch.epfl.bluebrain.nexus.delta.sourcing.query.StreamingQuery
+import doobie._
+import doobie.implicits._
+import io.circe.Json
+import monix.bio.{Task, UIO}
+
+trait GraphAnalyticsStream {
+
+  /**
+    * Allows to generate a non-terminating [[GraphAnalyticsResult]] stream for the given project
+    * @param project
+    *   the project to stream from
+    * @param start
+    *   the offset to start with
+    */
+  def apply(project: ProjectRef, start: Offset): ElemStream[GraphAnalyticsResult]
+
+}
+
+object GraphAnalyticsStream {
+
+  private val emptyMapType = Map.empty[Iri, Set[Iri]]
+
+  /**
+    * We look for a resource with these ids in this project and return their types if it can be found
+    */
+  private def query(project: ProjectRef, xas: Transactors)(nel: NonEmptyList[Iri]): UIO[Map[Iri, Set[Iri]]] = {
```

Comment:
`queryTypesOfResources`?

Reply:
It is a private method and the comment is enough no ?

Reply:
For it to be easy to read for someone else I think the method name needs more description. For someone who has never seen this class before it will really help